### PR TITLE
Fixed bug updating and creating user password in backend #1328

### PR DESCRIPTION
**Pull request by antonioperic**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | #1328
| License       | MIT
| Doc PR        | -

We need to use FosUserManager in order to update or create user with proper password

### CODE REVIEW

**Comment by stloyd** [comment]
In overall looks good, but that code duplication is crazy... we really need those Domain Managers or Managers + Factories approach...

@pjedrzejewski ping...

**Comment by antonioperic** [comment]
@stloyd  I agree duplication is bad, but we need then to find some other way to use UserManager in this scenario

**Comment by stloyd** [comment]
@pjedrzejewski ping & pong...

File: src/Sylius/Bundle/CoreBundle/Controller/UserController.php [hunk 1]
```diff
@@ -21,8 +25,150 @@ class UserController extends ResourceController
      */
     public function filterFormAction(Request $request)
     {
-        return $this->render('SyliusWebBundle:Backend/User:filterForm.html.twig', array(
-            'form' => $this->get('form.factory')->createNamed('criteria', 'sylius_user_filter', $request->query->get('criteria'))->createView()
-        ));
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/CoreBundle/Controller/UserController.php]: Does it make more readable? Not really. :) I'd extract it to separate method if needs change at all.

**Comment by pjedrzejewski** [comment]
How about listener?

**Comment by antonioperic** [comment]
Listener?

**Comment by pjedrzejewski** [comment]
Call the updateUser method in a listener, instead of overriding the controller. :)

**Comment by pjedrzejewski** [comment]
Actually, we have a handy listener here - https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/EventListener/UserUpdateListener.php#L24 Would be enough to update the user there with fos user manager.

**Comment by antonioperic** [comment]
will check that

**Comment by pjedrzejewski** [comment]
Replaced by and fixed in #2444.

**Pull request closed without merging**